parse.js

```diff
@@ -609,6 +621,47 @@ var parse = function (buf) {
     })
   })
 
+  schema.messages.forEach(function (msg) {
+    msg.fields.forEach(function (field) {
+      if (field.options && field.options.packed === 'true') {
+        if (PACKABLE_TYPES.indexOf(field.type) === -1) {
+          // let's see if it's an enum
+          if (field.type.indexOf('.') === -1) {
+            if (msg.enums && msg.enums.some(function (en) {
+              return en.name === field.type
```

Comment:
could you move the function out of the if? it's hard to understand whats going on here